src/main/java/com/microsoft/azure/cosmosdb/kafka/connect/source/CosmosDBSourceTask.java

```diff
@@ -70,32 +106,58 @@ public void start(Map<String, String> map) {
         logger.info("Started CosmosDB source task.");
     }
 
+    private JsonNode getLeaseContainerRecord() {
+        String sql = String.format("SELECT * FROM c WHERE c.Owner = '%s'", config.getWorkerName());
+        Iterable<JsonNode> filteredDocs = leaseContainer.queryItems(sql, new CosmosQueryRequestOptions(), JsonNode.class).toIterable();
```

Comment:
so this is getting the last info that this worker owned. what if the connector restarts, does it have the same worker name? 
can another worker come online and pickup where worker left off?

Reply:
Ah good catch, I'll rework the logic here.